portal-impl/src/com/liferay/portlet/expando/util/ExpandoBridgeIndexerImpl.java

```diff
@@ -157,50 +173,68 @@ else if (type == ExpandoColumnConstants.GEOLOCATION) {
 				fieldName.concat("_geolocation"), latitude, longitude);
 		}
 		else if (type == ExpandoColumnConstants.INTEGER) {
-			document.addKeyword(fieldName, expandoValue.getInteger());
+			document.addNumberSortable(fieldName, expandoValue.getInteger());
 		}
 		else if (type == ExpandoColumnConstants.INTEGER_ARRAY) {
 			if (!defaultValue) {
-				document.addKeyword(fieldName, expandoValue.getIntegerArray());
+				int[] intergerValues = expandoValue.getIntegerArray();
+
+				Integer[] integerArray = new Integer[intergerValues.length];
+
+				for (int i = 0; i < integerArray.length; i++) {
+					integerArray[i] = new Integer(intergerValues[i]);
+				}
+
+				document.addNumberSortable(fieldName, integerArray);
 			}
 			else {
-				document.addKeyword(fieldName, new int[0]);
+				document.addNumberSortable(fieldName, new Float[0]);
 			}
 		}
 		else if (type == ExpandoColumnConstants.LONG) {
-			document.addKeyword(fieldName, expandoValue.getLong());
+			document.addNumberSortable(fieldName, expandoValue.getLong());
 		}
 		else if (type == ExpandoColumnConstants.LONG_ARRAY) {
 			if (!defaultValue) {
-				document.addKeyword(fieldName, expandoValue.getLongArray());
+				document.addNumberSortable(fieldName,
+					ArrayUtil.toLongArray(expandoValue.getLongArray()));
 			}
 			else {
-				document.addKeyword(fieldName, new long[0]);
+				document.addNumberSortable(fieldName, new Long[0]);
 			}
 		}
 		else if (type == ExpandoColumnConstants.NUMBER) {
-			document.addKeyword(
-				fieldName, String.valueOf(expandoValue.getNumber()));
+			document.addNumberSortable(
+				fieldName, expandoValue.getNumber().longValue());
 		}
 		else if (type == ExpandoColumnConstants.NUMBER_ARRAY) {
 			if (!defaultValue) {
-				document.addKeyword(
+				document.addNumberSortable(
 					fieldName,
-					ArrayUtil.toStringArray(expandoValue.getNumberArray()));
+					ArrayUtil.toLongArray(expandoValue.getNumberArray()));
 			}
 			else {
-				document.addKeyword(fieldName, new long[0]);
+				document.addNumberSortable(fieldName, new Long[0]);
 			}
 		}
 		else if (type == ExpandoColumnConstants.SHORT) {
-			document.addKeyword(fieldName, expandoValue.getShort());
+			document.addNumberSortable(fieldName,
+				new Long(expandoValue.getShort()));
 		}
 		else if (type == ExpandoColumnConstants.SHORT_ARRAY) {
 			if (!defaultValue) {
-				document.addKeyword(fieldName, expandoValue.getShortArray());
+				short[] shortValues = expandoValue.getShortArray();
```

Comment:
@joshuacords Why a short array translate to a Long array, I know there is not API for short array, why not using Integer array which is closer to short